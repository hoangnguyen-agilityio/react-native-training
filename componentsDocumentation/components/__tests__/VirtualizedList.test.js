import React from 'react';
import renderer from 'react-test-renderer';
import VirtualizedListComponent from '../VirtualizedList';
import { Text, VirtualizedList } from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<VirtualizedListComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('there are 50 items in the list but only render 4 items', () => {
  const tree = renderer.create(<VirtualizedListComponent />);
  const list = tree.root.findAllByType(VirtualizedList)[0];
  const items = tree.root.findAllByType(Text);

  expect(list.props.getItemCount()).toBe(50);
  expect(items).toHaveLength(4);
});
