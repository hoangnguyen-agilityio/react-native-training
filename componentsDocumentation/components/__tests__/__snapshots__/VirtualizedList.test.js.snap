// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  initialNumToRender={4}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Item 1
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Item 2
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Item 3
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "height": 150,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Item 4
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    />
  </View>
</RCTScrollView>
`;
